app/models/report_type.rb

```diff
@@ -54,12 +63,19 @@ def self.respond_to_missing?(meth, include_private = false)
     meth.to_s =~ /^r\d\d\d$/ || super
   end
 
-  def self.available_standard_reports
-    {
+  def self.available_standard_reports(correspondence_types=nil)
+    all_reports = {
       foi: self.standard.foi.order(:full_name),
       sar: self.standard.sar.order(:full_name),
       offender_sar: self.standard.offender_sar.order(:full_name)
     }
+    allowed_reports = []
+    self.get_report_categories(correspondence_types).each do | report_category |
+      if !REPORT_TYPE_CORRESPONDENCE_TYPE_MAP[report_category].nil?
+        allowed_reports |= all_reports[REPORT_TYPE_CORRESPONDENCE_TYPE_MAP[report_category]]
+      end 
```

Comment:
Is the intention here to ensure that `allowed_reports` has only unique values in it - the bitwise OR (`|=`) with assignment is not that common to see so just checking that its not meant to be `||=`?